lib/driver.js

```diff
@@ -467,6 +467,23 @@ export class XCUITestDriver extends BaseDriver {
       // ensure WDA gets our defaults instead of whatever its own might be
       await this.updateSettings(wdaSettings);
 
+      if(this.opts.mjpegServerPort) {
+        this.log.debug(
+          `MJPEG broadcasting requested, forwarding MJPEG server port ${this.opts.mjpegServerPort} ` +
+            `to local port ${this.opts.mjpegServerPort}`
+        );
+        try {
+          await DEVICE_CONNECTIONS_FACTORY.requestConnection(caps.udid, this.opts.mjpegServerPort, {
+            devicePort: this.opts.mjpegServerPort,
+            usePortForwarding: true,
+          });
+        } catch {
+          this.log.warn(
+            `Cannot forward the local port ${this.opts.mjpegServerPort} to ${this.opts.mjpegServerPort} ` +
+              `on the device ${caps.udid}.`,
+          );
+        }
+      }
       // turn on mjpeg stream reading if requested
       if (this.opts.mjpegScreenshotUrl) {
```

Comment:
it makes sense to move mjpeg-related stuff to a separate method to improve readability